mysite/code/services/AddonUpdater.php

```diff
@@ -73,26 +74,34 @@ public function update($clear = false, $limitAddons = null) {
 
 		// This call to packagist can be expensive. Requests are served from a cache if usePackagistCache() returns true
 		$cache = SS_Cache::factory('addons');
-		if($this->usePackagistCache() && $packages = $cache->load('packagist')) {
+		if ($this->usePackagistCache() && $packages = $cache->load('packagist')) {
 			$packages = unserialize($packages);
 		} else {
 			$packages = $this->packagist->getPackages();
 			$cache->save(serialize($packages), 'packagist');
 		}
 		$this->elastica->startBulkIndex();
 
-		foreach ($packages as $package) {
+		foreach ($packages as $package) { /** @var Packagist\Api\Result\Package $package */
+			$isAbandoned = (property_exists($package, 'abandoned') && $package->abandoned);
 			$name = $package->getName();
 			$versions = $package->getVersions();
 
-			if($limitAddons && !in_array($name, $limitAddons)) continue;
+			if ($limitAddons && !in_array($name, $limitAddons)) {
+				continue;
+			}
 
 			$addon = Addon::get()->filter('Name', $name)->first();
 
 			if (!$addon) {
 				$addon = new Addon();
 				$addon->Name = $name;
 				$addon->write();
+			} elseif ($isAbandoned) {
+				// Abandoned packages should be removed during updates
+				echo ' - Removing abandoned package: ' . $name, PHP_EOL;
+				$addon->delete();
+				continue;
```

Comment:
Edge case - package doesn't exist locally, first time it gets written then not deleted - would only be deleted the second time around.